src/measure_theory/constructions/borel_space.lean

```diff
@@ -428,6 +428,129 @@ lemma is_preconnected.measurable_set
   (h : is_preconnected s) : measurable_set s :=
 h.ord_connected.measurable_set
 
+lemma exists_dense_seq_lt {α : Type*} [nonempty α] [topological_space α] [separable_space α]
+  [linear_order α] [order_topology α] (hnlb : ∀ x : α, ∃ y, y < x) (x : α) :
+  ∃ n, dense_seq α n < x :=
+exists_dense_seq_mem α (hnlb x) is_open_Iio
+
+lemma Union_Ico_dense_seq {α : Type*} [nonempty α] [topological_space α] [separable_space α]
+  [linear_order α] [order_topology α] (hnlb : ∀ x : α, ∃ y, y < x) (x : α) :
+  (⋃ n, Ico (dense_seq α n) x) = Iio x :=
+begin
+  ext y,
+  rw [mem_Iio, mem_Union],
+  split,
+  { rintro ⟨n, hmem⟩,
+    exact hmem.2 },
+  { intro h,
+    obtain ⟨n, hn⟩ := exists_dense_seq_lt hnlb y,
+    refine ⟨n, le_of_lt hn, h⟩ }
+end
+
+lemma borel_eq_generate_Ico (α : Type*) [nonempty α] [topological_space α]
+  [second_countable_topology α] [linear_order α] [order_topology α] :
+  borel α = generate_from {S | ∃ l u, l < u ∧ Ico l u = S} :=
+begin
+  refine le_antisymm _ (generate_from_le _),
+  { rw borel_eq_generate_Iio,
```

Comment:
If you add a generic `is_topological_basis_Ioo` (worth having anyway), and `borel_eq_generate_Ioo := is_topological_basis_Ioo.borel_eq_generate_from`, then you won't need to consider cases here.